processfamily/__init__.py

```diff
@@ -381,9 +366,155 @@ def _handle_response_line(self, line):
 CPU_AFFINITY_STRATEGY_CHILDREN_ONLY = 1
 CPU_AFFINITY_STRATEGY_PARENT_INCLUDED = 2
 
-CHILD_COMMS_STRATEGY_NONE = 0
-CHILD_COMMS_STRATEGY_PIPES_CLOSE = 1
-CHILD_COMMS_STRATEGY_PROCESSFAMILY_RPC_PROTOCOL = 2
+
+class NoCommsStrategy(ChildCommsStrategy):
+    MONITOR_STDOUT = False
+    CAN_WAIT_FOR_TERMINATE = False
+
+    @staticmethod
+    def get_popen_streams(echo_std_err):
+        """Returns kwargs for stdin, stdout and stderr to pass to subprocess.Popen"""
+        return {"stdin": None, "stdout": None, "stderr": None}
+
+class ClosePipesCommsStrategy(ChildCommsStrategy):
+    def stop_child(self, end_time):
+        p = self._process_instance
+        logger.info("Closing stdin for process %r", self)
+        try:
+            p.stdin.close()
+        except Exception as e:
+            logger.warning("Failed to close child process input stream with PID %s: %s\n%s", self.pid, e, _traceback_str())
+        yield
+        yield
+
+
+class ProcessFamilyRPCProtocolStrategy(ChildCommsStrategy):
+    SENDS_STDOUT_RESPONSES = True
+
+    def monitor_child_startup(self, end_time):
+        """generator method to monitor process startup, with the first yield after sending a ping,
+        the next after receiving a response, and stopping after cleanup"""
+        response_id = str(uuid.uuid4())
+        try:
+            yield self._send_command_req(response_id, "wait_for_start")
+            response = self._wait_for_response(response_id, end_time - time.time())
```

Comment:
Hmmmm. Good point. This has changed from the existing code but only in terms of the actual arithmetic sequence used to calculate things, not the functional effect of using `time.time()`.
We shouldn't be affecting this module with `virtualtime`, but because this is a separate library, it's also slightly tricky to work out how we should do this better. My proposal would be to change `virtualtime` so that it provides a way of getting an unwrapped module: then you can do:

```python
try:
    from virtualtime import realtime as time
except ImportError:
    import time
```
But perhaps that could go into a separate merge request then